src/test/isolation2/sql/workfile_gp_toolkit.sql

```diff
@@ -0,0 +1,33 @@
+-- This test checks for correct output of gp_toolkit.gp_workfile_entries view
+-- It is placed in integration tests because testing for this requires checking
+-- for workfiles while a query is still running.
+
+-- check there are no workfiles
+1: select segid, prefix, slice from gp_toolkit.gp_workfile_entries order by segid;
+
+1: create table workfile_test(id serial, s text) distributed by (id);
+1: insert into workfile_test(s) select v::text from generate_series(1, 2000) v;
+
+1: select gp_inject_fault('after_workfile_mgr_create_set', 'suspend', '', '', '', 2, 2, 0, dbid) from gp_segment_configuration where content > -1 and role = 'p';
+
+1&: select * from
+    (select * from workfile_test t1, workfile_test t2 order by t1.id+t2.id limit 10000000) x,
+    (select * from workfile_test t3, workfile_test t4 order by t3.id+t4.id limit 10000000) y,
+    generate_series(1, 10) z;
+-- wait until 2 workfile is created on each segment
+2: select gp_wait_until_triggered_fault('after_workfile_mgr_create_set', 1, dbid) from gp_segment_configuration where content > -1 and role = 'p';
+-- there should be exactly 6 workfiles, two for each segment (no duplication)
+2: select count(*) from gp_toolkit.gp_workfile_entries group by segid;
+
+-- interrupt the query
+2: select gp_inject_fault('create_function_fail', 'panic', dbid) from gp_segment_configuration where role='p' and content = 0;
+2: create function my_function() returns void as $$ begin end; $$ language plpgsql;
```

Comment:
Isn't it too hard for a saw file test?

Reply:
maybe
```suggestion
2: SELECT pg_cancel_backend(pid) FROM pg_stat_activity WHERE query LIKE '%workfile_test%' and pid != pg_backend_pid();
```

Reply:
We have to stop the query from finishing (because it runs for a very long time), and there are no useful fault injectors that are guaranteed to hit no matter how far the query has progressed. It is easier to just cause a segment panic to kill the query backend. This is an isolation test so there are no other tests running in parallel, so there should be no unintended consequences.

Reply:
Works, fixed